python/py_pkg_cc_deps.bzl

```diff
@@ -1,100 +1,148 @@
 "Repository rule exporting C-language dependencies contained in a Python package."
 
-# This rule extends the standard @rules_python rules to expose C-language dependences
-# contained in some Python packages like NumPy. It extends @rules_python rather
-# than duplicating the download mechanism, and to ensure the versions of Python
-# packages used throughout the WWORKSPACE are consistent. This
-# could conceivably become a built-in feature of @rules_python at some point.
-
-def _find_headers(ctx, pkg, include_prefix):
-    # Find the path to the headers within the @rules_python repository for the Python
+# This extends the standard rules_python rules to expose C-language dependences
+# contained in some Python packages like NumPy. It extends rules_python to
+# avoid duplicating the download mechanism, and to ensure the Python package
+# versions used throughout the WWORKSPACE are consistent.
+
+def _rules_python_path(ctx, pkg):
+    # Make an absolute path to the rules_python repository for the Python
     # package `pkg`.
 
     # WARNING: To get a flesystem path via ctx.path(), its argument must be a
     # label to a non-generated file. ctx.path() does not work on non-file
     # A standard technique for finding the path to a repository (see,
     # e.g., rules_go) is to use the repository's BUILD file; however, the exact
-    # name of the build file is an implementation detail of @rules.python.
-    build = pkg.relative(":BUILD.bazel")
-    path = ctx.path(build).dirname
+    # name of the build file is an implementation detail of rules_python.
+    build_file = pkg.relative(":BUILD.bazel")
+    abspath = ctx.path(build_file).dirname
+    return abspath
+
+def _join_paths(a, b):
+    if type(a) == "string" and type(b) == "string":
+        result = "/".join((a, b))
+
+    elif type(a) == "path" and type(b) == "string":
+        # Append components of string b to path a, because path.get_child()
+        # requires one component at a time.
+        result = a
+        for x in b.split("/"):
+            result = result.get_child(x)
 
-    # rules_python unpacks the Python package within the site-packages directory.
-    path = path.get_child("site-packages")
+    return result
 
-    # Append the include_prefix, if any. get_child() needs one component at a
-    # time.
-    if include_prefix:
-        for c in include_prefix.split("/"):
-            path = path.get_child(c)
+def _make_build_file(basedir, include_paths, libs):
```

Comment:
let's add to the PR description a brief set of steps to inspect the BUILD file that gets created for tensorflow-cpu as a result of this rule.

Reply:
Added in the PR description and in in-line comments in the implementation in 8f2073a.

Rather than give steps, I suggested finding and examining the repository directory created in the cache when debugging the build or implementation. Giving steps would have been awkward considering the context, and probably unnecessarily detailed.